### PR TITLE
feat(ecs): Override Task Def Artifacts

### PR DESCRIPTION
**Pull request by paragbhingre**
Add a support to override task definitions.

Frontend changes to support [#6374](https://github.com/spinnaker/spinnaker/issues/6374)

Related PRs - 
Clouddriver - [#5314](https://github.com/spinnaker/clouddriver/pull/5314)
ORCA - [#4117](https://github.com/spinnaker/orca/pull/4117)

Screen Shot 2021-04-15 at 5 23 50 PM

### CODE REVIEW

File: app/scripts/modules/ecs/src/serverGroup/configure/wizard/taskDefinition/TaskDefinition.tsx [hunk 1]
```diff
@@ -342,6 +352,15 @@ export class TaskDefinition extends React.Component<ITaskDefinitionProps, ITaskD
             />
           </div>
         </div>
+
+        <StageConfigField groupClassName="form-group evaluateTaskDef" label="Expression Evaluation" helpKey="ecs.evaluateExpression" labelColumns={5} fieldColumns={7} >
+          <CheckboxInput
+            checked={command.evaluateTaskDefinitionArtifactExpressions === true}
+            onChange={() =>  { updateEvaluateTaskDefArtifactFlag(); }}
+            text="Evaluate Task Definition Artifact"
```
Comment (allisaurus) [inline app/scripts/modules/ecs/src/serverGroup/configure/wizard/taskDefinition/TaskDefinition.tsx:360]: be more explicit here: "Evaluate SpEL expressions in artifact" (I think we can omit "task definition" b/c that's the section of the wizard we're in here)

**Review by allisaurus — COMMENTED** [review]
LGMT, save some editorial changes.

File: test/functional/cypress/integration/ecs/server_group.spec.js [hunk 2]
```diff
@@ -213,4 +213,28 @@ describe('amazon ecs: ECSApp Server Group', () => {
     cy.get('[data-test-id="ServerGroupWizard.submitButton"]').click();
     cy.get('[data-test-id="Pipeline.revertChanges"]').click();
   });
+
+  it('edit an existing server group to enable SpEL processoing for task def artifact', () => {
```
[the remark this answers is not in the document]
Comment (paragbhingre) [reply test/functional/cypress/integration/ecs/server_group.spec.js:217]: Oops, corrected

File: app/scripts/modules/ecs/src/ecs.help.ts [hunk 3]
```diff
@@ -83,6 +83,8 @@ const helpContents: { [key: string]: string } = {
     '<p>Designates how many tasks, at a minimum, to run on the specified capacity provider. Only one capacity provider in a capacity provider strategy can have a <em>base</em> defined.</p>',
   'ecs.capacityProviderWeight':
     '<p>Designates the relative percentage of the total number of tasks launched that should use the specified capacity provider.</p>',
+  'ecs.evaluateExpression':
+    '<p>SpEL expression evaluation of the task definition artifact in this stage. Following checkbox let\'s you evaluate your task definition artifact coming from external sources.(e.g. GitHub) </p>',
```
[the remark this answers is not in the document]
Comment (paragbhingre) [reply app/scripts/modules/ecs/src/ecs.help.ts:87]: Updated the content as per ask

File: app/scripts/modules/ecs/src/serverGroup/configure/wizard/taskDefinition/TaskDefinition.tsx (hunk 1 above)
Comment (paragbhingre) [reply app/scripts/modules/ecs/src/serverGroup/configure/wizard/taskDefinition/TaskDefinition.tsx:360]: make sense

File: app/scripts/modules/ecs/src/ecs.help.ts [hunk 4]
```diff
@@ -83,6 +83,8 @@ const helpContents: { [key: string]: string } = {
     '<p>Designates how many tasks, at a minimum, to run on the specified capacity provider. Only one capacity provider in a capacity provider strategy can have a <em>base</em> defined.</p>',
   'ecs.capacityProviderWeight':
     '<p>Designates the relative percentage of the total number of tasks launched that should use the specified capacity provider.</p>',
+  'ecs.evaluateExpression':
+    '<p>Whether to evaluate <a href="https://spinnaker.io/guides/user/pipeline/expressions/"><b>pipeline expressions</b></a> within the task definition artifact in this stage. Checking this box let\'s you evaluate your task definition artifact coming from external sources.(e.g. GitHub) </p>',
```
Comment (allisaurus) [inline app/scripts/modules/ecs/src/ecs.help.ts:87]: please add `target="_blank"` within the leading `<a>` tag to ensure the link opens in a new tab (see line 78 for example)

**Review by allisaurus — APPROVED** [review]
LGTM, save a small HTML change (also please rebase from master). 
Thanks for this @paragbhingre ! Approving but will hold off merging until the related PRs are similarly approved.

**Pull request merged**